Makefile

```diff
@@ -53,18 +88,35 @@ PROTOC_OS = linux
 PROTOC_ARCH := $(UNAME_ARCH)
 endif
 
-PROTOC := $(CACHE)/protoc-$(PROTOC_VERSION).zip
-$(PROTOC):
+PROTOC_DOWNLOAD := $(CACHE)/protoc-$(PROTOC_VERSION).zip
+$(PROTOC_DOWNLOAD):
 	@if ! command -v curl >/dev/null 2>/dev/null; then echo "error: curl must be installed" >&2; exit 1; fi
 	@if ! command -v unzip >/dev/null 2>/dev/null; then echo "error: unzip must be installed" >&2; exit 1; fi
-	@rm -f $(BIN)/protoc
 	$(eval PROTOC_TMP := $(shell mktemp -d))
 	curl -sSL https://github.com/protocolbuffers/protobuf/releases/download/v$(PROTOC_VERSION)/protoc-$(PROTOC_VERSION)-$(PROTOC_OS)-$(PROTOC_ARCH).zip -o $(PROTOC_TMP)/protoc.zip
-	@mkdir -p $(BIN)
-	unzip -q $(PROTOC_TMP)/protoc.zip -d $(dir $(BIN)) bin/protoc
+	@mkdir -p $(dir $@)
+	@mv $(PROTOC_TMP)/protoc.zip $@
 	@rm -rf $(PROTOC_TMP)
+
+$(PROTOC): $(PROTOC_DOWNLOAD)
+	@mkdir -p $(BIN)
+	unzip -DD -q $(PROTOC_DOWNLOAD) -d $(dir $(BIN)) bin/protoc
+	chmod u+w $@
+
+CONNECT_CONFORMANCE_DOWNLOAD := $(CACHE)/connect-conformance-$(CONFORMANCE_VERSION).tgz
+$(CONNECT_CONFORMANCE_DOWNLOAD):
+	@if ! command -v curl >/dev/null 2>/dev/null; then echo "error: curl must be installed" >&2; exit 1; fi
+	@if ! command -v tar >/dev/null 2>/dev/null; then echo "error: tar must be installed" >&2; exit 1; fi
+	$(eval CONFORMANCE_TMP := $(shell mktemp -d))
+	curl -sSL https://github.com/connectrpc/conformance/releases/download/$(CONFORMANCE_VERSION)/connectconformance-$(CONFORMANCE_VERSION)-$(UNAME_OS)-$(UNAME_ARCH).tar.gz -o $(CONFORMANCE_TMP)/conformance.tgz
 	@mkdir -p $(dir $@)
-	@touch $@
```

Comment:
This seemed weird that it would create an empty placeholder zip file. So I tweaked this so that the artifact it creates is the actual downloaded ZIP file.

I also added a distinct target for the `.tmp/bin/protoc` binary itself because, TBH, I found it very confusing that the `$(PROTOC)` target was actually referencing a bogus ZIP file.

Reply:
At some point I'd love to move all of this into the gradle build so we only have to maintain a single build system but this is a nice improvement.